Make InternalModel and Resnet work with rexportable flow

tugsbayasgalan opened the pull request:
Summary: Internal model and Resnet uses "re-export" flow now. Also did some refactoring to make the code little cleaner

Some changes for OSS:
1. Correctly use the "cached" fake tensors so that static symbols are still resolved to static 
2. Change logic in PassBase to allocate static shapes for parameters 
3. Add "is_torch_exported" tag to every node to make it survive during various graph transformations.
4. Added experimental wrapper API for quantization team to get pre_dispatch=True graph. Note that it doesn't actually do that right now. But we plan to switch soon. 

Test Plan: CI

Differential Revision: D47890878

jerryzh168 commented on torch/_export/__init__.py:
```diff
@@ -120,6 +127,18 @@ class ExportDynamoConfig:
 DECOMP_TABLE = core_aten_decompositions()
 
 
+def capture_pre_autograd_graph(
```
can you add some docs about plans for this API

tugsbayasgalan replied:
just added

tugsbayasgalan committed:
Make Resnet work with rexportable flow (#106676)

Test Plan: CI

Differential Revision: D47890878

fbshipit-source-id: 50f0e12f60f6a37dab5e30d22122882594fd6227

jerryzh168 reviewed (approved):
quant changes looks good to me, do you need review from some compiler people as well?

tugsbayasgalan commented:
> quant changes looks good to me, do you need review from some compiler people as well?

Sounds good! I will merge after I get review from Zhengxu.
